runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -293,14 +299,396 @@ int32_t TR::ARM64PrivateLinkage::buildPrivateLinkageArgs(TR::Node *callNode,
+               }
+            numIntegerArgs++;
+            break;
+         case TR::Float:
+            argRegister = pushFloatArg(child);
+            if (numFloatArgs < numFloatArgRegs)
+               {
+               if (!cg()->canClobberNodesRegister(child, 0))
+                  {
+                  tempReg = cg()->allocateRegister(TR_FPR);
+                  generateTrg1Src1Instruction(cg(), TR::InstOpCode::fmovs, callNode, tempReg, argRegister);
+                  argRegister = tempReg;
+                  }
+               if (numFloatArgs == 0 && resType.isFloatingPoint())
+                  {
+                  TR::Register *resultReg;
+                  if (resType.getDataType() == TR::Float)
+                     resultReg = cg()->allocateSinglePrecisionRegister();
+                  else
+                     resultReg = cg()->allocateRegister(TR_FPR);
+                  dependencies->addPreCondition(argRegister, TR::RealRegister::v0);
+                  dependencies->addPostCondition(resultReg, TR::RealRegister::v0);
+                  }
+               else
+                  TR::addDependency(dependencies, argRegister, properties.getFloatArgumentRegister(numFloatArgs), TR_FPR, cg());
+               }
+            else // numFloatArgs >= numFloatArgRegs
+               {
+               mref = getOutgoingArgumentMemRef(argMemReg, argRegister, TR::InstOpCode::vstrposts, pushToMemory[argIndex++]);
+               }
+            numFloatArgs++;
+            break;
+         case TR::Double:
+            argRegister = pushDoubleArg(child);
+            if (numFloatArgs < numFloatArgRegs)
+               {
+               if (!cg()->canClobberNodesRegister(child, 0))
+                  {
+                  tempReg = cg()->allocateRegister(TR_FPR);
+                  generateTrg1Src1Instruction(cg(), TR::InstOpCode::fmovd, callNode, tempReg, argRegister);
+                  argRegister = tempReg;
+                  }
+               if (numFloatArgs == 0 && resType.isFloatingPoint())
+                  {
+                  TR::Register *resultReg;
+                  if (resType.getDataType() == TR::Float)
+                     resultReg = cg()->allocateSinglePrecisionRegister();
+                  else
+                     resultReg = cg()->allocateRegister(TR_FPR);
+                  dependencies->addPreCondition(argRegister, TR::RealRegister::v0);
+                  dependencies->addPostCondition(resultReg, TR::RealRegister::v0);
+                  }
+               else
+                  TR::addDependency(dependencies, argRegister, properties.getFloatArgumentRegister(numFloatArgs), TR_FPR, cg());
+               }
+            else // numFloatArgs >= numFloatArgRegs
+               {
+               mref = getOutgoingArgumentMemRef(argMemReg, argRegister, TR::InstOpCode::vstrpostd, pushToMemory[argIndex++]);
+               }
+            numFloatArgs++;
```

Comment:
The double case is nearly identical to the float case except for the `pushFloatArg()/pushDoubleArg()` call, and the two opcodes for the generated instructions.  Can you try and combine these cases in some way to reduce the code size?

Reply:
Merged the `TR::Double` case with the `TR::Float` case.